[BUGFIX LTS] ensure “pause on exception” pauses in the right place

stefanpenner opened the pull request:
It is quite common (like in ember-data’s tests) to not have a global onError handler in tests.
But because of the previous state of code, we would still end up paused in the default dispatchError rather then where the actual error was thrown.

This addresses the issue, but ensuring onErrorTarget.onerror returns undefined if no onError has been set

before:

<img width="946" alt="screen shot 2017-08-25 at 6 15 44 am" src="https://user-images.githubusercontent.com/1377/29716219-6885d146-895f-11e7-838d-f45f4e3b3f21.png">

After: 

<img width="943" alt="screen shot 2017-08-25 at 6 16 01 am" src="https://user-images.githubusercontent.com/1377/29716228-6d5b9c8c-895f-11e7-86b9-213e97cd5e57.png">

stefanpenner committed:
[BUGFIX LTS] ensure “pause on exception” pauses in the right place

It is quite common (like in ember-data’s tests) to not have a global onError handler in tests.
But because of the previous state of code, we would still end up paused in the default dispatchError rather then where the actual error was thrown.

This addresses the issue, but ensuring onErrorTarget.onerror returns undefined if no onError has been set

rwjblue commented on packages/ember-metal/tests/error_handler_test.js:
```diff
@@ -0,0 +1,66 @@
+import Ember from 'ember';
```
This test should likely be moved to the `packages/ember/tests` folder, since it requires the global now.

krisselden commented:
```Test failed: Template scoping examples: global failure
    Failed assertion: TypeError: undefined is not an object (evaluating '_ember.default.Test.adapter')```

Commit:
[BUGFIX release] Update backburner.js to 1.2.1.

The diff between [v1.1.0 and
v1.2.0](https://github.com/BackburnerJS/backburner.js/compare/v1.1.0...v1.2.1)
includes the following changes:

* Allow `onError` to be late bound (and recalculated for each flush).
* Make `now()` late bound (allowing for things like
  `sinon.useFakeTimers()`).

Commit:
[BUGFIX beta] Cleanup test to ensure `backburner._platform` is present.

Previously, this test assumed that the only value on
`backburner._platform` that was required was `setTimeout`.  As of
Backburner 1.2.1 this is no longer true (it also requires `.now()`), and
really should not have been assumed to begin with.

This change simply ensures that `backburner._platform` continues to have
all the same methods on it, but uses the test override of `setTimeout`
to run its assertions.

Commit:
[BUGFIX lts] Fix test issues exposed by not swallowing errors.

Previously, due to always running within a try/catch these errors were
swallowed in production build test runs.

rwjblue commented:
Updated:

* rebase
* update backburner.js to 1.2.1 (to include other fixes required)
* fix tests that were previously throwing errors that were swallowed in production
* fix test that was clobbering `backburner._platform`

I tested both prod and debug builds before pushing, so now we just have to convince sauce labs 😈 ...

stefanpenner commented:
thanks @rwjblue
